### DISCUSSION TITLE
What's the easiest way to deploy my API?

### DISCUSSION THREAD

**Discussion by gavinmcfarland** [vercel/vercel#4503] [Help] [answered]
I've built an application which runs an express server to give access to an API. I'd like to deploy this using Now. Looking at the documentation I can use the `api` folder but this requires me to use Now's own server similar to express, which means I have to rewrite my API.

How can I use Now to deploy my own API which runs on its own express server?

**Comment by wwwflorencio** [comment, marked as answer]
Hi bro, for this case you need to create the `now.json` file config in the project (root folder) with this content:

```json
{
  "version": 2,
  "public": true,
  "builds": [
    {
      "src": "server.js",
      "use": "@now/node-server",
      "config": {}
    }
  ],
  "routes": [
    {
      "headers": {
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET, POST, PUT, DELETE, OPTIONS",
        "Access-Control-Allow-Headers": "X-Requested-With, Content-Type, Accept"
      },
      "src": "/.*",
      "dest": "/server.js"
    }
  ]
}
```

If your main API file is not the `server.js` just rename this on config (`dest` section).

Then, use the now command:
`now`

**Reply by gavinmcfarland** [reply]
Thank you kindly! I'll give this a go.

**Reply by gavinmcfarland** [reply]
Thanks this works. I have an issue when trying to restrict it to a subpath, but it must be an issue between my app and Now though, because it works with a simple hello server.

The following works:

__now.json__
```
{
	"version": 2,
	"public": true,
	"builds": [
		{
			"src": "server.js",
			"use": "@now/node-server",
			"config": {
				"includeFiles": [ "content/**" ]
			}
		}
	],
	"routes": [
		{
			"headers": {
				"Access-Control-Allow-Origin": "/api/*",
				"Access-Control-Allow-Methods": "GET, POST, PUT, DELETE, OPTIONS",
				"Access-Control-Allow-Headers": "X-Requested-With, Content-Type, Accept"
			},
			"src": "/api/.*",
			"dest": "/server.js"
		}
	]
}
```

__server.js__
```
var express = require('express');

const app = express();

app.get('/api/', (req, res) => {
	res.send('Hello World!');
});

app.listen('3000', () => {
	console.log(`Server listening at http://localhost:3000`);
});
```

But when I use it with my own app I get a blank page.

**Reply by wwwflorencio** [reply]
Nice :D
